tests/Test.Yaapii.Xambly/Directive/AddIfChildDirectiveTests.cs

```diff
@@ -53,5 +53,33 @@ public void DoesntAddTwice()
                 ).Apply(new XDocument()).ToString()
             );
         }
+
+        [Fact]
+        public void IsCaseSensitive()
+        {
+            Assert.Equal(
+                "<root>\r\n  <item>\r\n    <id>some-id</id>\r\n    <child />\r\n  </item>\r\n  <item>\r\n    <id>SOME-ID</id>\r\n    <child />\r\n  </item>\r\n</root>",
+                new Xambler(
+                    new ManyOf<IDirective>(
+                        new PushDirective(),
+                        new AddDirective("root"),
+                        new PushDirective(),
+                        new AddDirective("item"),
+                        new AddDirective("id"),
+                        new SetDirective("some-id"),
+                        new PopDirective(),
+                        new AddIfChildDirective("item", "id", "some-id"),
+                        new AddDirective("child"),
+                        new UpDirective(),
+                        new UpDirective(),
+                        new AddIfChildDirective("item", "id", "SOME-ID", false),
```

Comment:
i think this test would have been green with the previous version (without the false), wouldn't it?
i think what you want to test is:
```
Assert.False(
    ...
    new AddIfChildDirective("item", "id", "some-ID", false),
    ...
```

Reply:
No. The trick here is that we have two `<item>`s with different ids. Without passing `false` here, the second would not have been created and both `<child>`s would have been created in the first `<item>` with `<id>some-id</id>` (lowercase).

Reply:
understood now, thanks 👍